### PR TITLE
Bump Serving, Eventing to 1.4 for 1.25

### PR DESCRIPTION
**Pull request by skonto**
## Proposed Changes

- As per title.
- As in https://github.com/openshift-knative/serverless-operator/pull/1602
- Split in separate commits for easier review.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -177,14 +178,11 @@ replace (
 	github.com/go-logr/zapr => github.com/go-logr/zapr v0.4.0
 	github.com/manifestival/manifestival => github.com/manifestival/manifestival v0.7.0
 
-	k8s.io/api v0.23.4 => k8s.io/api v0.22.5
```
Comment (skonto) [inline go.mod:180]: Keeping the replacements here:

```
	k8s.io/api v0.23.5 => k8s.io/api v0.22.5
	k8s.io/apimachinery v0.23.5 => k8s.io/apimachinery v0.22.5
	k8s.io/client-go v0.23.5 => k8s.io/client-go v0.22.5
```
causes
```
$ ./hack/update-deps.sh 
go: finding module for package k8s.io/client-go/kubernetes/typed/flowcontrol/v1beta2
go: finding module for package k8s.io/api/flowcontrol/v1beta2
go: finding module for package k8s.io/client-go/applyconfigurations/autoscaling/v2
go: finding module for package k8s.io/api/autoscaling/v2
go: finding module for package k8s.io/client-go/kubernetes/typed/autoscaling/v2
go: finding module for package k8s.io/client-go/applyconfigurations/flowcontrol/v1beta2
go: found k8s.io/api/autoscaling/v2 in k8s.io/api v0.24.3
go: found k8s.io/api/flowcontrol/v1beta2 in k8s.io/api v0.24.3
go: found k8s.io/client-go/applyconfigurations/autoscaling/v2 in k8s.io/client-go v0.24.3
go: found k8s.io/client-go/applyconfigurations/flowcontrol/v1beta2 in k8s.io/client-go v0.24.3
go: found k8s.io/client-go/kubernetes/typed/autoscaling/v2 in k8s.io/client-go v0.24.3
go: found k8s.io/client-go/kubernetes/typed/flowcontrol/v1beta2 in k8s.io/client-go v0.24.3
```

**Comment by skonto** [comment]
```
kafka-storage-version-migrator
    postupgrade.go:102: jobs.batch "knative-kafka-storage-version-migrator" not found 
        job:
         {"metadata":{"creationTimestamp":null},"spec":{"template":{"metadata":{"creationTimestamp":null},"spec":{"containers":null}}},"status":{}}
```

**Comment by pierDipi** [comment]
testing this potential fix https://github.com/openshift-knative/serverless-operator/pull/1654/commits/c9b2421cd792a234a4cd607ef7197a701c999c96

**Comment by skonto** [comment]
In PR #1648 the test was [skipped](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1648/pull-ci-openshift-knative-serverless-operator-main-4.10-upgrade-tests-aws-ocp-410/1549334033303867392/artifacts/upgrade-tests-aws-ocp-410/serverless-e2e/build-log.txt). 

> PASS test/upgrade

=== Skipped
=== SKIP: test/upgrade TestServerlessUpgrade/PostUpgradeTests/BrokerPostUpgradeTest/Verify_post-install (0.00s)
    postupgrade.go:64: SO uses different names
            --- SKIP: TestServerlessUpgrade/PostUpgradeTests/BrokerPostUpgradeTest/Verify_post-install (0.00s)

=== SKIP: test/upgrade TestServerlessUpgrade/PostUpgradeTests/SinkPostUpgradeTest/Verify_post-install (0.00s)
    postupgrade.go:64: SO uses different names
            --- SKIP: TestServerlessUpgrade/PostUpgradeTests/SinkPostUpgradeTest/Verify_post-install (0.00s)

**Comment by pierDipi** [comment]
Ah, you're right, we miss a patch in eventing-kafka-broker v1.4

**Comment by skonto** [comment]
@pierDipi pls let me know when the patch lands there so I can re-fetch the dep and try again.

**Review by pierDipi — APPROVED** [review]
/lgtm

**Comment by skonto** [comment]
Tested a bit on 4.6.x, some results [here](https://gist.github.com/skonto/4c835305b3e1aa063f099d19b4764bfe).
Didn't notice anything related to the k8s api update, will also verify via the 4.6 periodic jobs. 
/unhold

**Pull request merged**